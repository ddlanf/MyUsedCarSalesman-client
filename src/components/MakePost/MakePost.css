
.make-post{
    margin-top: 11%;
}

.make-post-heading{
    font-size: 7vw;
    text-align: center;
    margin-top: 5%;
    margin-bottom: 6%;
    margin-left: auto;
    margin-right: auto;
    padding-right: 3%;
    font-family: 'Merriweather Sans', sans-serif;
}

.make-post-form{
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 10%;
    padding-bottom: 2%;
    position: relative;
    background-color: #EAEDEC;
    border: #999090 solid 2px;
    border-radius: 10%;
}

.make-post-input-box{
    width: 100%;
    margin-left: auto;
    margin-right: auto;
    padding-left: 8%;
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: stretch;
}


.make-post-right, .make-post-left{
    width: 100%;
    min-width: 250px;
    margin: 0;
}

textarea{
    margin-left: 0%;
    position: relative;
    top: -0.2%;
}

.make-post-label{
    width: 100%;
    display: block;
    font-size: 4vw;
}

.make-post-input{
    display: block;
    margin-bottom: 3%;
    width: 30%;
    font-size: 4vw;
}

 .make-post-commission{
    width: 70%;
}

.make-post-description{
    height: 20vw;
    width: 70%;
 }

 .make-post-terms-and-cond{
    width: 70%;
    height: 20vw;
 }

 .make-post-location-box{
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
    width: 80%;
 }

.make-post-image{
    width: 55%;

 }

 .make-post-submit{
    display: block;
    height: 5vw;
    width: 40%;
    height: 10vw;
    max-height: 60px;
    max-width: 300px;
    border-radius: 10%;
    color: black;
    margin-top: 4%;
    margin-bottom: 2%;
    margin-left: auto;
    margin-right: auto;
    font-size: 5vw;
    cursor: pointer;
}

.make-post-error, .make-post-buffer{
    font-size: 3vw;
    text-align: center;
    border: 1px solid #999090;
    background-color: #DFE32E;
    width: 50%;
    margin-top: 0;
    margin-bottom: 2%;
    margin-left: auto;
    margin-right: auto;
    overflow-x: auto;
}

.make-post-error{
    color: red;
}

.make-post-buffer{
    color: black;
}

@media screen and (min-width: 1000px) {

    .make-post-heading{
        font-size: 60px;
        margin-bottom: 0;
        margin-top: 4%;
    }

    .make-post-right, .make-post-left{
        width: 50%;
    }

    .make-post-input-box{
        width: 100%;
        padding-left: 10%;
    }

    .make-post-label{
        font-size: 35px;
    }
    
    .make-post-input{
        width: 50%;
        font-size: 35px;
    }

    .make-post-commission{
        width: 70%;
    }
    
    .make-post-description{
        width: 70%;
     }
    
     .make-post-terms-and-cond{
        width: 70%;
     }
    
    .make-post-image{
        width: 50%;
     }

     .make-post-submit{
        margin-top: 2%;
        font-size: 40px;
    }
}