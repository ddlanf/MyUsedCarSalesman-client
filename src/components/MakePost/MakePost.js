import React, { Component } from 'react'
import './MakePost.css'
export default class MakePost extends Component {
    render() {
        return (
            <div className="make-post">
                <h1 className="make-post-heading">Create Post</h1>
                <section className="make-post-form-box">
                    <form className="make-post-form">
                        <div className="make-post-input-box">
                            <div className="make-post-left">
                                <label name="make" className="make-post-label">Make</label>
                                <input type="text" className="make-post-input"></input>
                                <label name="model" className="make-post-label">Model</label>
                                <input type="text" className="make-post-input"></input>
                                <label name="year" className="make-post-label">Year</label>
                                <input type="text" className="make-post-input"></input>
                                <label name="mileage" className="make-post-input">Mileage</label>
                                <input type="text" className="make-post-input"></input>
                                <label name="description" className="make-post-label">Description</label>
                                <textarea className="make-post-description make-post-input"></textarea>
                            </div>
                            <div className="make-post-right">
                                <label name="price" className="make-post-label">Price</label>
                                <input type="text" className="make-post-input"></input>
                                <label name="commission" className="make-post-label">Commision</label>
                                <input className="make-post-input make-post-commission"></input>
                                <label name="location" className="make-post-label">Location</label>
                                <input className="make-post-input make-post-location" ></input>
                                <label name="description" className="make-post-label">Other Terms and Conditions</label>
                                <textarea className="make-post-terms-and-cond make-post-input"></textarea>
                                <label htmlFor="img" className="img make-post-label">Image Link:</label>
                                <input className="make-post-input make-post-image" ></input>
                            </div>
                        </div>
                        <button id="submit" type="submit" className="make-post-submit">Submit</button>
                    </form>
                </section> 
            </div>
        )
    }
}
