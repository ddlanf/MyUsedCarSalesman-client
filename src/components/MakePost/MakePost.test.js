import React from 'react';
import ReactDOM from 'react-dom';
import MakePost from './MakePost';
import renderer from 'react-test-renderer';

describe('MakePost component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MakePost/>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<MakePost/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});