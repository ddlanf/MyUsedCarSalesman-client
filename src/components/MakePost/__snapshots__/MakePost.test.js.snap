// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakePost component renders the UI as expected 1`] = `
<div
  className="make-post"
>
  <section
    className="make-post-form-box"
  >
    <form
      className="make-post-form"
      onSubmit={[Function]}
    >
      <h1
        className="make-post-heading"
      >
        Create Post
      </h1>
      <div
        className="make-post-input-box"
      >
        <div
          className="make-post-left"
        >
          <label
            className="make-post-label"
            name="make"
          >
            Make
          </label>
          <input
            className="make-post-input"
            name="make"
            onChange={[Function]}
            type="text"
          />
          <label
            className="make-post-label"
            name="model"
          >
            Model
          </label>
          <input
            className="make-post-input"
            name="model"
            onChange={[Function]}
            type="text"
          />
          <label
            className="make-post-label"
            name="year"
          >
            Year
          </label>
          <input
            className="make-post-input"
            name="year"
            onChange={[Function]}
            type="number"
          />
          <label
            className="make-post-label"
            name="mileage"
          >
            Mileage
          </label>
          <input
            className="make-post-input"
            name="mileage"
            onChange={[Function]}
            type="number"
          />
          <label
            className="make-post-label"
            name="description"
          >
            Description
          </label>
          <textarea
            className="make-post-description make-post-input"
            name="description"
            onChange={[Function]}
          />
          <label
            className="img make-post-label"
            htmlFor="img"
            name="image"
          >
            Image Link:
          </label>
          <input
            className="make-post-input make-post-image"
            name="image"
            onChange={[Function]}
          />
        </div>
        <div
          className="make-post-right"
        >
          <label
            className="make-post-label"
            name="price"
          >
            Price
          </label>
          <input
            className="make-post-input"
            name="price"
            onChange={[Function]}
            type="number"
          />
          <label
            className="make-post-label"
            name="commission"
          >
            Commision (Ex: 20%)
          </label>
          <input
            className="make-post-input make-post-commission"
            name="commission_amount"
            onChange={[Function]}
          />
          <label
            className="make-post-label"
            name="location"
          >
            City
          </label>
          <input
            className="make-post-input make-post-city"
            name="city"
            onChange={[Function]}
            type="text"
          />
          <label
            className="make-post-label"
            name="location"
          >
            State
          </label>
          <input
            className="make-post-input make-post-state"
            name="state"
            onChange={[Function]}
            type="text"
          />
          <label
            className="make-post-label"
            name="terms-and-cond"
          >
            Other Terms and Conditions
          </label>
          <textarea
            className="make-post-terms-and-cond make-post-input"
            name="other_terms_and_conditions"
            onChange={[Function]}
          />
        </div>
      </div>
      
      <button
        className="make-post-submit"
        disabled={false}
        id="submit"
        style={
          Object {
            "backgroundColor": "#999090",
          }
        }
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</div>
`;
